File: api/src/main/java/de/hipp/pnp/api/fivee/interfaces/FiveECharacterProducer.java
package de.hipp.pnp.api.fivee.interfaces;

import com.fasterxml.jackson.core.JsonProcessingException;
import de.hipp.pnp.api.fivee.abstracts.BaseCharacter;

import java.util.List;

public interface FiveECharacterProducer {
    BaseCharacter generate(int gameType) throws JsonProcessingException;

    List<BaseCharacter> getAllCharacters();
}

File: api/src/main/java/de/hipp/pnp/api/fivee/interfaces/FiveECharacterProducer.kt
package de.hipp.pnp.api.fivee.interfaces

import com.fasterxml.jackson.core.JsonProcessingException
import de.hipp.pnp.api.fivee.abstracts.BaseCharacter

interface FiveECharacterProducer {
    @Throws(JsonProcessingException::class)
    fun generate(gameType: Int): String?

    fun allCharacters(): MutableList<BaseCharacter?>?
}
